src/modules/differential_drive_control/DifferentialDriveControl.hpp

```diff
@@ -91,15 +99,35 @@ class DifferentialDriveControl : public ModuleBase<DifferentialDriveControl>, pu
 	uORB::Subscription _parameter_update_sub{ORB_ID(parameter_update)};
 	uORB::Subscription _vehicle_control_mode_sub{ORB_ID(vehicle_control_mode)};
 	uORB::Subscription _vehicle_status_sub{ORB_ID(vehicle_status)};
+	uORB::Subscription _global_pos_sub{ORB_ID(vehicle_global_position)};
+	uORB::Subscription _pos_sp_triplet_sub{ORB_ID(position_setpoint_triplet)};
+	uORB::Subscription _vehicle_attitude_sub{ORB_ID(vehicle_attitude)};
+	uORB::Subscription _vehicle_angular_velocity_sub{ORB_ID(vehicle_angular_velocity)};
+	uORB::Subscription _vehicle_local_position_sub{ORB_ID(vehicle_local_position)};
+
 
 	uORB::PublicationMulti<actuator_motors_s> _actuator_motors_pub{ORB_ID(actuator_motors)};
 	uORB::Publication<differential_drive_setpoint_s> _differential_drive_setpoint_pub{ORB_ID(differential_drive_setpoint)};
 
 	differential_drive_setpoint_s _differential_drive_setpoint{};
-	DifferentialDriveKinematics _differential_drive_kinematics{};
-
+	position_setpoint_triplet_s _pos_sp_triplet{};
+	vehicle_attitude_s _vehicle_attitude{};
+	vehicle_global_position_s _global_pos{};
+	vehicle_angular_velocity_s _vehicle_angular_velocity{};
+	vehicle_local_position_s _vehicle_local_position{};
 	bool _armed = false;
 	bool _manual_driving = false;
+	bool _mission_driving = false;
```

Comment:
Its because of the external ros2 modes, if one is active, it will also publish and get subscribed to from this line:

https://github.com/PX4/PX4-Autopilot/blob/3c53b1f0e824bf94506e7f8e5228748264cc4895/src/modules/differential_drive_control/DifferentialDriveControl.cpp#L170-L171

And if the mission and the external mode is running, they will interfere with eachother.

Reply:
okay but do you distinguish between mission and other auto modes? Otherwise I would call it `_auto_driving`

Reply:
true, thats a good point, I will call it _auto_driving! Thanks :)